### DISCUSSION TITLE
have a sort of xml / txt / whatever file for a users beatmaps

### DISCUSSION THREAD

**Discussion by velamyGH** [ppy/osu#27927] [Ideas]
my idea is basically a way to import, export, and generally store what maps a user has on their client to their profile for ease of backing up and restoring for other installs. a map in this list can be marked as "not downloaded" if said file is moved to a new install, and a user can be prompted to download it either from song select (i.e, updating the map as it would be empty without downloading) or the current direct! page. this can be indicated similarly to the "local" and "pending, ranked, etc" statuses. i just think this would be nice with lazer being supported on so many devices, as well as this method not taking up more bandwidth than necessary. hope im making my idea clear, but i can explain my idea further if needed :)

**Comment by peppy** [comment]
You can copy the installation folder to a new PC and it will work fine.

**Reply by velamyGH** [reply]
yes, but the point is to not have to transfer over a billion maps and save time and even data this way since youll only have to copy over a couple mb (this is especially true for mobile devices) and if this could be tied to the users account then there would be no copying needed in the first place

**Reply by peppy** [reply]
You mean you want to download gigabytes from our servers every time rather than copy the data yourself between PCs?

**Reply by velamyGH** [reply]
my idea was the maps wouldnt auto download this way, like i mentioned it would just mark them as "not downloaded" based on the file and give you the option to download the map if youd want. it wouldnt download more than what someone would already want to play right?

**Reply by velamyGH** [reply]
okay, i do now see the issues with players just downloading everything on the list regardless and wasting bandwidth. maybe some sort of built in wifi transfer would be better for things like mobile devices? it would only happen over the users home network but i totally get if this is niche enough to not be worth the effort and low demand

**Comment by cihe13375** [comment]
I recall that some time ago I run into a weird situation (somewhat on my own fault though) where the content of many files on my drive are override by garbage bits, but the filenames are not changed. I recovered my stable song folder by (programmatically) extracting all beatmapsetid from the subfolder names, and batch downloaded them from mirror sites.

I don't know if it's possible to do something similar in lazer. Maybe very rare scenario anyway though.
